consensus-types/types/attestation_data.go

```diff
@@ -79,13 +79,8 @@ func (a *AttestationData) MarshalSSZ() ([]byte, error) {
 	return buf, ssz.EncodeToBytes(buf, a)
 }
 
-// NewFromSSZ creates a new AttestationData object from SSZ format.
-func (a *AttestationData) NewFromSSZ(buf []byte) (*AttestationData, error) {
-	if a == nil {
-		a = &AttestationData{}
-	}
-	return a, ssz.DecodeFromBytes(buf, a)
-}
```

Comment:
This and other methods are replaced by a single, generic function, `sszconstructors.NewFromSSZ`.

Horrible name I reckon, and suggestions are welcomed.
I really really like the reduced code duplication